### PR TITLE
Histogram of number of merge operands

### PR DESCRIPTION
**Pull request by siying**
Summary: Add a histogram in statistics to help users understand how many merge operands they merge.

Test Plan: Observe the stats in db_bench

### CODE REVIEW

File: include/rocksdb/statistics.h [hunk 1]
```diff
@@ -370,6 +370,9 @@ enum Histograms : uint32_t {
   BYTES_DECOMPRESSED,
   COMPRESSION_TIMES_NANOS,
   DECOMPRESSION_TIMES_NANOS,
+  // Number of merge operands passed to the merge operator in user read
```
Comment (ajkr) [inline include/rocksdb/statistics.h]: does it also include merges during compactions?

Comment (siying) [inline include/rocksdb/statistics.h]: From my reading the code, it doesn't. Maybe I'm wrong. Let me double check that.

**Comment by siying** [comment]
No. It's for the Cassandra experiments.

File: include/rocksdb/statistics.h (hunk 1 above)
Comment (siying) [inline include/rocksdb/statistics.h]: Hmm. I tested it. Maybe I'm wrong. Let me investigate more and make sure merges done in compactions are ruled out.

**Comment by siying** [comment]
Updated to address @ajkr 's comment about the compaction merges. Good catch!

**Comment by sagar0** [comment]
There are few more usages of `TimedFullMerge` in merge_helper.cc, write_batch.cc and get_context.cc. I am curious, why did you exclude counting stats from them ie. `update_num_ops_stats=false`(by default) in those places.

**Comment by siying** [comment]
@sagar0 I want to measure the performance impact to user queries. Background compactions may not necessarily impact user performance.

**Pull request closed without merging**